task/host-sp-comms/src/bsp/cosmo_a.rs

```diff
@@ -102,17 +102,16 @@ impl ServerImpl {
             //
             // Sharkfin connectors start at J200 and are numbered sequentially
             4..=13 => {
-                let (designator, f): ([u8; 4], _) =
+                let (designator, f): ([u8; 7], _) =
                     Self::get_sharkfin_vpd(index as usize - 4);
                 let mut name = *b"____/U2/ID";
-                name[0..4].copy_from_slice(&designator);
+                name[0..4].copy_from_slice(&designator[0..4]);
```

Comment:
Nit: this would be more clear if we overwrote the lower 7 bytes with the whole designator, e.g.

```rust
let mut name = *b"_______/ID";
name[0..7].copy_from_slice(&designator);
```